### ISSUE TITLE
Target matching bug in findListener

### ISSUE THREAD

**Issue by ntodd** [adopted-ember-addons/ember-file-upload#159]
The change made in 5d1639e1c412998a7b2ed94a44bc2a1c15286eda to fix an IE bug appears to cause the `dragenter` listener to match incorrectly on the `<body>` tag. The `selector` is a descendant of `<body>` so the finder `evt.target.contains(element)` will always return true any time the event target is `<body>` (any time the dragged item is over the body element).

Undoing the above commit appears to restore correct behavior on latest Chrome and Firefox. I am unaware of the original IE issue, but this does not appear to be a correct cross-browser fix.

**Comment by jvann** [comment]
I am having the same problem, were you able to fix it @ntodd ?

**Comment by jvann** [comment]
I tried reverting the commit, but it did not work, were you able to find another solution @ntodd ?

**Comment by schleifer-john** [comment]
I did some more debugging of the `findListener` method.  The issue here is that `evt.target` behaves differently in some browsers, like IE and Safari, and returns parent div tags instead of the div tag of the `file-dropzone` component.  Without the `evt.target.contains(element)`, condition, the drag & drop functionality does not work in these browsers.
See: https://github.com/tim-evans/ember-file-upload/issues/141#issuecomment-383218931
and https://github.com/tim-evans/ember-file-upload/issues/136

I'm not sure of a way to fix the `findListener` method right now that doesn't use `evt.target`.  I did come across some articles of how to get drag & drop working with jquery.  The code to set up the listeners looked similar to what is being done here, but there may be a different way to set this up that sidesteps the `findListener` issue.

Alternatively, I attempted to convert the drag & drop event listening logic to leverage the Ember Event Names described in the documentation here: https://guides.emberjs.com/release/components/handling-events/#toc_event-names

This refactoring worked fine in Chrome, but the issue I am running into is that the `drop`, `dragOver`, and `dragLeave` events are not being triggered in Internet Explorer 11.  The only event that is triggered in IE is `dragEnter`.  I'm not sure if this is a bug in Ember, but for whatever reason these other events are not working.

**Comment by schleifer-john** [comment]
I went back down the path of refactoring the component to leverage the Ember Event Names.  I was able to get the other events to trigger in IE 11 after I added `evt.preventDefault()` to the `dragEnter` event.

I have this working in my local project, but I need to copy the code over to my fork and write some test cases before I do a PR.  Unfortunately, I have a deadline coming up soon, so it may be a while before I can do this.

In the meantime, here is the workaround:
1. Generate a new component using Ember CLI, I called mine 'file-dropzone-ex': `ember g component file-dropzone-ex`
2. For the Component Template file, copy the code in the 'file-dropzone' component template file here: https://github.com/tim-evans/ember-file-upload/blob/latest/addon/components/file-dropzone/template.hbs
3. For the Component Javascript file, copy and paste the contents of this gist: https://gist.github.com/schleifer-john/44dda865e7b3a504dcfbfc09960ec640
